src/attention/utils/helpers.ts

```diff
@@ -163,20 +164,31 @@ export async function useRecompute(state: AttentionState) {
     middleware: [
       offset({ mainAxis: state?.distance ?? 8, crossAxis: state?.skidding ?? 0}), // offers flexibility over how to place the attentionEl towards its targetEl both on the x and y axis (horizontally and vertically).
       state?.flip && flip({ //when flip is set to true it will move the attentionEl's placement to its opposite side or to the preferred placements if fallbackPlacements has a value
-        fallbackAxisSideDirection: 'start', // the preferred placement axis fit when flip is set to true and fallbackPlacements does not have a value. 'start' represents 'top' or 'left'.
+        crossAxis: state?.crossAxis, //checks overflow to trigger a flip. When disabled, it will ignore overflow
         fallbackPlacements: state?.fallbackPlacements,
       }),
-      !state?.flip && autoPlacement(), // when flip is set to false, it will instead call autoPlacement() that will choose the placement that has the most space available automatically. You can only use either flip() or autoPlacement(), and not combined. 
-      shift({ padding: 16}),
       !state?.noArrow && state?.arrowEl && arrow({ element: state?.arrowEl }),
+      hide({ //will hide the attentionEl when it appears detached from the targetEl. Can be called multiple times in the middleware-array if you want to use several strategies
+        strategy: 'escaped', //default strategy is 'referenceHidden'
+      }),
+      hide(), // we call hide() again here to also trigger the 'referenceHidden' strategy. 
     ],
   }).then(({ x, y, middlewareData, placement }) => {
     state.actualDirection = placement
-    
+
     Object.assign(attentionEl?.style, {
       left: `${x}px`,
       top: `${y}px`,
-    })
+    });
+
+    if (middlewareData?.hide && !state?.isTooltip) {      
```

Comment:
If I understand correctly, this condition should apply for any Attention that is visible and absolutely positioned (popover, tooltip, highlight + `isShowing`)? If so, shouldn't this check for `!isCallout` instead?

Reply:
I added `!state?.isTooltip` because I noticed when I added this styling to hide the attentionEl when targetEl is hidden the attentionEl should get `visibility: hidden` and when targetEl is visible, attentionEl should instead get `visibility: visible`, which in that case made the tooltip-visible even when you were no longer hovering on the targetEl. However, I realize now that it should be enough for me to change it to this instead: 
```
if (middlewareData?.hide) {      
      const { escaped, referenceHidden } = middlewareData?.hide
      Object.assign(attentionEl?.style, {
        visibility: referenceHidden ? 'hidden' : '',
        opacity: escaped ? '0.5' : '',
      })
    }
```

I don't think it makes a difference if I add `!state.isCallout`, since we don't use targetEl for the callout. But I could add it just as a safety? :)

Reply:
Updated in this commit: [bd92181](https://github.com/warp-ds/core/pull/16/commits/bd921812ce77f2a7efa1defab2558259821f525f)